petl/io/xlsx.py

```diff
@@ -86,15 +87,32 @@ def __iter__(self):
             pass
 
 
-def toxlsx(tbl, filename, sheet=None, write_header=True):
+def toxlsx(tbl, filename, sheet=None, write_header=True, overwrite=True):
```

Comment:
Nice improvement! :+1: 

Some options here:

1. Maybe change `overwrite` from `True`/`False` to:
    1. `overwrite` the file
    2. `replace` the sheet content in the file. If the `sheet` argument is `None` recreate the file.
    3. `append` to the sheet content  the file
2. Perhaps the default should be `overwrite='replace'` for reducing compatibility problems.
    1. Calls with argument `sheet` with `None` will behave as currently.
    2. Calls with  argument `sheet` filled with constant names by multiple `jobs` will have some impact by only a fixed number of sheets.
    3. Calls with  argument `sheet` filled with random names by multiple `jobs` will be impacted and will keep growing. However this usage may not be common.

Reply:
Sounds good. It feels a bit weird to replace a single sheet if one is named, and the entire file if not, but it does make sense. However, `replace` with no sheet name also replaces `overwrite` because they are entirely identical, so the latter isn't needed.

Calling the operation that adds sheets to existing files `append` invites confusion with `appendxlsx()` that adds rows to existing sheets, so I'd prefer to call it `add`.

Updated PR coming up, either today or tomorrow.